apis/server/exec_bridge.go

```diff
@@ -55,8 +56,13 @@ func (s *Server) startContainerExec(ctx context.Context, rw http.ResponseWriter,
 	name := mux.Vars(req)["name"]
 	_, upgrade := req.Header["Upgrade"]
 
+	if err := s.ContainerMgr.CheckExecExist(ctx, name); err != nil {
```

Comment:
I am wondering if we should keep consistent that object existence checking should be located in the manager layer rather than bridge layer?

So maybe we should encapsulate this checking into the function `StartExec`.

Reply:
if check in `StartExec`, it too late. `StartExec` will not return error, since hikacked connection not take error back, we only write error message into connect, then api will get the error message, if check is in StartExec, we will never get error.